http-svr/xmlrpc_client.ml

```diff
@@ -96,6 +98,64 @@ let get_new_stunnel_id =
   let m = Mutex.create () in
   fun () -> Mutex.execute m (fun () -> incr counter; !counter)
 
+let run_watchdog timeout (fire_fn : unit -> unit) f (pout,pin) =
+  let fired = ref None in
+  let th = Thread.create (fun () ->
+    match Unix.select [pout] [] [] timeout with
```

Comment:
select can wake up early for a variety of reasons with EINTR, especially if the process is receiving signals. It would be good to use a function here that sleeps based on an absolute deadline, pthread_ and sem_ have a few but I haven't seem any of them bound in ocaml standard library.
Thought that Thread.wait_timed_read would DTRT, but it just calls select without retrying.

Might have to use gettimeofday, compute how long is left and retry ourselves unless we can find a helper function that does that already

Reply:
Perhaps the solution here helps: https://caml.inria.fr/mantis/print_bug_page.php?bug_id=793

Reply:
EINTR has a more dramatic effect on xapi: https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi.ml#L974-L989